### PR TITLE
Blob <-> Bitstring casting 

### PR DESCRIPTION
**Pull request by LindsayWray**
In the updated list of fusser issues #5984  some bitstring issues popped up (no.47 and no 60)This PR addresses these. 
When casting a BIT type to/from BLOB, a ReinterpretCast is not sufficient, the first byte containing metadata about the bit string needs to be set or removed explicitly.

### CODE REVIEW

File: src/function/cast/bit_cast.cpp [hunk 1]
```diff
@@ -10,7 +10,7 @@ BoundCastInfo DefaultCasts::BitCastSwitch(BindCastInput &input, const LogicalTyp
 		// bit to varchar
 		return BoundCastInfo(&VectorCastHelpers::StringCast<string_t, duckdb::CastFromBit>);
 	case LogicalTypeId::BLOB:
-		return DefaultCasts::ReinterpretCast;
+		return BoundCastInfo(&VectorCastHelpers::StringCast<string_t, duckdb::CastBitToBlob>);
```
Comment (Mytherin) [inline src/function/cast/bit_cast.cpp:13]: Should we support bit <-> blob casts at all? It seems that Postgres does not support them:

```sql
postgres=# select cast('01010101' as bit(8))::bytea;
ERROR:  cannot cast type bit to bytea
LINE 1: select cast('01010101' as bit(8))::bytea;
                                         ^
postgres=# SELECT cast(bytea '\xDEADBEEF' as bit(100));
ERROR:  cannot cast type bytea to bit
LINE 1: SELECT cast(bytea '\xDEADBEEF' as bit(100));

```

File: extension/parquet/column_writer.cpp [hunk 2]
```diff
@@ -1265,7 +1265,11 @@ class StringColumnWriter : public BasicColumnWriter {
 
 			if (validity.RowIsValid(vector_index)) {
 				run_length++;
-				const auto &value = strings[vector_index];
+				auto &value = strings[vector_index];
+
+				if (vector.GetType().id() == LogicalTypeId::BIT) {
+					value = string_t(value.GetDataUnsafe() + 1, value.GetSize() - 1); // skip first byte of bitstring
```
Comment (Mytherin) [inline extension/parquet/column_writer.cpp:1271]: I wonder if it perhaps makes more sense to convert bits to strings when writing to Parquet?

Alternatively we could leave this unsupported for now as well.

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good - some comments:

File: src/function/cast/bit_cast.cpp (hunk 1 above)
Comment (LindsayWray) [reply src/function/cast/bit_cast.cpp]: I had initially added it to accommodate adding bit type to test_all_types for parquet, but indeed there seems no real reason to support this, so I've removed it altogether!

File: extension/parquet/column_writer.cpp (hunk 2 above)
Comment (LindsayWray) [reply extension/parquet/column_writer.cpp]: Dito, I've taken it out!

**Comment by PedroTadim** [comment]
And now the bit type is not available in the C-API. I just updated the list.

**Pull request merged**